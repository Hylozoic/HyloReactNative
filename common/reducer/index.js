import { LOGIN, LOGOUT } from '../components/Login/actions'
import { CHECK_SESSION } from '../components/SessionCheck/actions'

export default function rootReducer (state, action) {
  const { type, error, payload, meta } = action

  if (error) {
    switch (type) {
      case LOGIN:
        return {
          ...state,
          loginError: payload.message || payload.response.body
        }
    }
    return
  }

  switch (type) {
    case LOGIN:
      return {
        ...state,
        loginError: null,
        loggedIn: true,
        defaultLoginEmail: meta.email // TODO persist this
      }
    case CHECK_SESSION:
      return {
        ...state,
        loggedIn: payload
      }
    case LOGOUT:
      return {
        ...state,
        loggedIn: false
      }
  }
  return {}
}
