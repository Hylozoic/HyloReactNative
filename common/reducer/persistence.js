import { AsyncStorage } from 'react-native'
export const PERSISTED_STATE_KEY = 'persisted-app-state-v1'

export function persist (reducer) {
  return function (state, action) {
    const nextState = reducer(state, action)
    if (nextState !== state) {
      const serialized = JSON.stringify(nextState)
      const startTime = Date.now()
      AsyncStorage.setItem(PERSISTED_STATE_KEY, serialized, () => {
        console.log(`persisted state in ${Date.now() - startTime}ms`)
      })
    }
    return nextState
  }
}
