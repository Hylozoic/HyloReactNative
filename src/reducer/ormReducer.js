import { EXTRACT_MODEL } from '../store/constants'
import orm from '../store/models'
import ModelExtractor from './ModelExtractor'

export default function ormReducer (state = {}, action) {
  const session = orm.session(state)
  const { payload, type, meta, error } = action
  if (error) return state

  switch (type) {
    case EXTRACT_MODEL:
      ModelExtractor.addAll({
        session,
        root: payload,
        modelName: meta.modelName,
        append: meta.append
      })
      break
  }

  return session.state
}
