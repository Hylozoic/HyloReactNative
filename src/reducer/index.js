import { LOGIN, LOGIN_WITH_FACEBOOK, LOGOUT } from '../components/Login/actions'
import { FETCH_CURRENT_USER } from '../store/actions/fetchCurrentUser'
import { CHECK_SESSION } from '../components/SessionCheck/actions'
import { persist } from './persistence'
import { omit } from 'lodash/fp'

function rootReducer (state = {}, action) {
  const { type, error, payload, meta } = action

  if (error) {
    switch (type) {
      case LOGIN:
      case LOGIN_WITH_FACEBOOK:
        return {
          ...state,
          loginError: payload.message || payload.response.body
        }
    }
    return
  }

  switch (type) {
    case LOGIN:
      return {
        ...omit('loginError', state),
        loggedIn: true,
        defaultLoginEmail: meta.email
      }
    case LOGIN_WITH_FACEBOOK:
      return {
        ...omit('loginError', state),
        loggedIn: true
      }
    case CHECK_SESSION:
      if (payload !== state.loggedIn) {
        return {...state, loggedIn: payload}
      }
      return state
    case LOGOUT:
      return {
        ...state,
        loggedIn: false
      }
    case FETCH_CURRENT_USER:
      return {
        ...state,
        currentUser: payload.data.me
      }
  }

  return state
}

export default persist(rootReducer)
