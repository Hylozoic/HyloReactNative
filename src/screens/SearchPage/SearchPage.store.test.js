import reducer, {
  setSearchTerm,
  setSearchFilter,
  getSearchTerm,
  getSearchFilter,
  fetchSearchResults,
  defaultState,
  MODULE_NAME
} from './SearchPage.store'

describe('reducer, setters and getters', () => {
  describe('on setSearchTerm', () => {
    it('sets the search term', () => {
      const search = 'lai'
      const action = setSearchTerm(search)
      const state = {
        [MODULE_NAME]: reducer(defaultState, action)
      }
      expect(getSearchTerm(state)).toEqual(search)
    })
  })
  describe('on setSearchFilter', () => {
    it('sets the search term', () => {
      const filter = 'post'
      const action = setSearchFilter(filter)
      const state = {
        [MODULE_NAME]: reducer(defaultState, action)
      }
      expect(getSearchFilter(state)).toEqual(filter)
    })
  })
})

describe('fetchSearchResults', () => {
  it('matches snapshot', () => {
    const params = {
      search: 'lae',
      offset: 20,
      filter: 'comment'
    }
    expect(fetchSearchResults(params)).toMatchSnapshot()
  })
})
