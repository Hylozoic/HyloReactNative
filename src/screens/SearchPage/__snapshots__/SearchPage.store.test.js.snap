// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchSearchResults matches snapshot 1`] = `
{
  "graphql": {
    "query": "query ($search: String, $type: String, $offset: Int) {
        search(term: $search, first: 10, type: $type, offset: $offset) {
          total
          hasMore
          items {
            id
            content {
              __typename
              ... on Person {
                id
                name
                location
                avatarUrl
                skills {
                  items {
                    id
                    name
                  }
                }
              }
              ... on Post {
                
  id
  announcement
  title
  details
  type
  creator {
    id
    name
    avatarUrl
    groupRoles {
      name
      emoji
      active
      groupId
    }
    moderatedGroupMemberships {
      groupId
    }
  }
  createdAt
  updatedAt
  isAnonymousVote
  isPublic
  fulfilledAt
  startTime
  endTime
  donationsLink
  projectManagementLink
  myEventResponse
  commenters(first: 20) {
    id
    name
    avatarUrl
  }
  commentersTotal
  commentsTotal
  
  linkPreview {
    description
    id
    imageUrl
    title
    url
  }
  linkPreviewFeatured
  location
  locationObject {
    id
    addressNumber
    addressStreet
    bbox {
      lat
      lng
    }
    center {
      lat
      lng
    }
    city
    country
    fullText
    locality
    neighborhood
    region
  }
  peopleReactedTotal
  proposalStatus
  proposalOutcome
  votingMethod
  quorum
  proposalOptions {
    total
    hasMore
    items {
      id
      text
      emoji
    }
  }
  proposalVotes {
      total
      hasMore
      items {
        id
        optionId
        user {
          id
          name
          avatarUrl
        }
      }
  }
  myReactions {
    emojiFull
    id
  }
  postReactions {
    emojiFull
    id
    user {
      id
      name
    }
  }
  groups {
    id
    name
    slug
  }
  attachments {
    type
    url
    position
    id
  }
  postMemberships {
    id
    pinned
    group {
      id
    }
  }
  topics {
    id
    name
    postsTotal
    followersTotal
  }
  members {
    total
    hasMore
    items {
      id
      name
      avatarUrl
      bio
      tagline
      location
    }
  }
  eventInvitations {
    total
    hasMore
    items {
      id
      response
      person {
        id
        name
        avatarUrl
        bio
        tagline
        location
      }
    }
  }

              }
              ... on Comment {
                id
                text
                createdAt
                creator {
                  id
                  name
                  avatarUrl
                }
                post {
                  id
                  title
                  type
                  creator {
                    id
                    name
                    avatarUrl
                  }
                }
                attachments {
                  id
                  url
                  type
                }
              }
            }
          }
        }
      }",
    "variables": {
      "offset": 20,
      "search": "lae",
      "type": "comment",
    },
  },
  "meta": {
    "extractModel": "SearchResult",
    "extractQueryResults": {
      "getItems": [Function],
    },
  },
  "type": "SearchPage/FETCH_SEARCH",
}
`;
